api/envoy/extensions/filters/common/dependency/v3/dependency.proto

```diff
@@ -0,0 +1,37 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.common.dependency.v3;
+
+import "udpa/annotations/migrate.proto";
+import "udpa/annotations/status.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.common.dependency.v3";
+option java_outer_classname = "DependencyProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Filter dependency specification]
+
+// Dependency specification and string identifier.
+message Dependency {
+  enum DependencyType {
+    Header = 0;
+    FilterStateKey = 1;
+    DynamicMetadata = 2;
+  }
+
+  DependencyType type = 1;
+
+  string name = 2;
+}
+
+// Dependency specification for a filter.
+message FilterDependencies {
+  repeated Dependency decode_dependencies = 1;
+
+  repeated Dependency decode_providencies = 2;
```

Comment:
Leaning towards `decode_providers` etc. as something a bit clearer for those not in on the original design discussion.